plugin-dell/dputilities/taskService.go

```diff
@@ -0,0 +1,417 @@
+	connPool, err := common.GetDBConnection(common.InMemory)
+	if err != nil {
+		log.Error("error while trying to get the db connection")
+		return nil, fmt.Errorf("error while trying to connnect to DB: %v", err.Error())
+	}
+	taskData, err := connPool.Read("task", taskID)
+	if err != nil {
+		return nil, fmt.Errorf("error while trying to read from DB: %v", err.Error())
+	}
+
+	task, errs := ts.unmarshalTask([]byte(taskData))
+	if errs != nil {
+		return nil, fmt.Errorf("error while trying to unmarshal task data: %v", errs)
+	}
+
+	return task, nil
+}
+
+//BuildCompletedTaskIndex is used to build the index for Completed Task
+func buildCompletedTaskIndex(completedTask *Task, table string) error {
+	conn, err := common.GetDBConnection(common.InMemory)
+	if err != nil {
+		return fmt.Errorf("error while trying to connecting to DB: %v", err.Error())
+	}
+	key := completedTask.UserName + "::" + completedTask.EndTime.String() + "::" + completedTask.ID
+	createError := conn.CreateTaskIndex(CompletedTaskIndex, completedTask.EndTime.UnixNano(), key)
+	if createError != nil {
+		return fmt.Errorf("error while trying to create task index: %v", err)
+	}
+	return nil
+}
+
+func (ts *TaskServiceImpl) unmarshalTask(taskData []byte) (*Task, error) {
+	dbTask := new(TaskDb)
+	errorMsg := "error while trying to unmarshal task data: %v"
+	if err := json.Unmarshal(taskData, &dbTask); err != nil {
+		return nil, fmt.Errorf(errorMsg, err)
+	}
+
+	taskState, err := ts.GetTaskState(dbTask.TaskState)
+	if err != nil {
+		return nil, fmt.Errorf(errorMsg, err)
+	}
+
+	taskStatus, err := ts.GetTaskStatus(dbTask.TaskStatus)
+	if err != nil {
+		return nil, fmt.Errorf(errorMsg, err)
+	}
+
+	return &Task{
+		ParentID:        dbTask.ParentID,
+		ID:              dbTask.ID,
+		URI:             dbTask.URI,
+		UserName:        dbTask.UserName,
+		Name:            dbTask.Name,
+		HidePayload:     dbTask.HidePayload,
+		Payload:         dbTask.Payload,
+		PercentComplete: dbTask.PercentComplete,
+		TaskMonitor:     dbTask.TaskMonitor,
+		TaskState:       taskState,
+		TaskStatus:      taskStatus,
+		StatusCode:      dbTask.StatusCode,
+		TaskResponse:    dbTask.TaskResponse,
+		Messages:        dbTask.Messages,
+		StartTime:       dbTask.StartTime,
+		EndTime:         dbTask.EndTime,
+	}, nil
+}
+
+func PrepareDbConfig() {
+	config.Data.DBConf = &config.DBConf{
+		Protocol:             pluginConf.Data.DBConf.Protocol,
+		InMemoryHost:         pluginConf.Data.DBConf.InMemoryHost,
+		InMemoryPort:         pluginConf.Data.DBConf.InMemoryPort,
+		OnDiskHost:           pluginConf.Data.DBConf.OnDiskHost,
+		OnDiskPort:           pluginConf.Data.DBConf.OnDiskPort,
+		MaxIdleConns:         pluginConf.Data.DBConf.MaxIdleConns,
+		MaxActiveConns:       pluginConf.Data.DBConf.MaxActiveConns,
+		RedisHAEnabled:       pluginConf.Data.DBConf.RedisHAEnabled,
+		InMemorySentinelPort: pluginConf.Data.DBConf.InMemorySentinelPort,
+		InMemoryMasterSet:    pluginConf.Data.DBConf.InMemoryMasterSet,
```

Comment:
Update OnDiskSentinelPort and OnDiskMasterSet paramaters too, as common.CheckDBConnection() checks for both instances of DB.

Reply:
Updated